core/src/main/java/org/elasticsearch/common/util/concurrent/AbstractRunnable.java

```diff
@@ -36,7 +36,7 @@ public final void run() {
         try {
             doRun();
         } catch (InterruptedException ex) {
-            Thread.interrupted();
+            Thread.currentThread().interrupt();
```

Comment:
As a base class it should stick to standard practice (which is to restore the flag). If subclasses want to do anything differently, they can always catch `InterruptedException` and handle it explicitly.